File: Data Structures & Algorithms/buy-and-sell-crypto/submission-5.java
class Solution {
    public int maxProfit(int[] prices) {
        int left = 0;
        int right = 0;
        int profit = 0;

        while(right < prices.length){
            if(prices[right]<=prices[left]){
                left = right;
            }
            profit = Math.max(profit, prices[right]-prices[left]);
            right+=1;
        }
        return profit;
    }
}

File: Data Structures & Algorithms/max-water-container/submission-5.py
class Solution:
    def maxArea(self, heights: List[int]) -> int:
        left = 0
        total = 0
        right = len(heights)-1
        while left < right:
            current = (min(heights[right], heights[left]) * (right-left))
            total = max(total, current)
            if heights[right] > heights[left]:
                left+=1
            else:
                right-=1
        return total


